webapp/src/main/java/com/box/l10n/mojito/rest/commit/CommitWS.java

```diff
@@ -0,0 +1,100 @@
+package com.box.l10n.mojito.rest.commit;
+
+import com.box.l10n.mojito.entity.Commit;
+import com.box.l10n.mojito.entity.Repository;
+import com.box.l10n.mojito.rest.View;
+import com.box.l10n.mojito.rest.repository.RepositoryWithIdNotFoundException;
+import com.box.l10n.mojito.service.commit.CommitService;
+import com.box.l10n.mojito.service.pullrun.PullRunWithNameNotFoundException;
+import com.box.l10n.mojito.service.pushrun.PushRunWithNameNotFoundException;
+import com.box.l10n.mojito.service.repository.RepositoryNameNotFoundException;
+import com.box.l10n.mojito.service.repository.RepositoryRepository;
+import com.fasterxml.jackson.annotation.JsonView;
+import org.joda.time.DateTime;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+import java.util.List;
+
+/**
+ * @author garion
+ */
+@RestController
+public class CommitWS {
+
+    /**
+     * logger
+     */
+    static Logger logger = LoggerFactory.getLogger(CommitWS.class);
+
+    @Autowired
+    RepositoryRepository repositoryRepository;
+
+    @Autowired
+    CommitService commitService;
+
+    @JsonView(View.Commit.class)
+    @RequestMapping(value = "/api/commits/", method = RequestMethod.GET)
+    public Commit getCommit(@RequestParam String commitName, @RequestParam Long repositoryId){
```

Comment:
I've added an additional getCommits() pageable version that takes in a variety of parameters - and kept getCommit under "/api/commits/detailed" to be able to also return additional information in that response (Push and Pull data) - that otherwise we wouldn't want to be joining on.